From 14599cd760203a43be992c50fa1dc6a4c6e473ea Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Wed, 4 Jan 2012 09:21:05 -0800
Subject: [PATCH] Fix live update script (#769970)

pushd and popd are not available in the shell used by dracut.
---
 modules.d/90dmsquash-live/apply-live-updates.sh |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/modules.d/90dmsquash-live/apply-live-updates.sh b/modules.d/90dmsquash-live/apply-live-updates.sh
index 8dce5d4..f840d1a 100755
--- a/modules.d/90dmsquash-live/apply-live-updates.sh
+++ b/modules.d/90dmsquash-live/apply-live-updates.sh
@@ -1,8 +1,9 @@
 #!/bin/sh
 if [ -b /dev/mapper/live-rw ]; then
-    if pushd /updates &>/dev/null; then
+    if [ -d /updates ]; then
         echo "Applying updates to live image..."
+        cd /updates
         /bin/cp -a -t $NEWROOT .
-        popd &>/dev/null
+        cd -
     fi
 fi
